src/librustdoc/html/markdown.rs

```diff
@@ -440,6 +502,23 @@ pub fn render(w: &mut fmt::Formatter,
                 Event::Start(Tag::Strong) => {
                     strong(parser, buffer, toc_builder, shorter, id);
                 }
+                Event::Start(Tag::Rule) => {
+                    rule(parser, buffer, toc_builder, shorter, id);
+                }
+                Event::Start(Tag::FootnoteDefinition(ref def)) => {
+                    let mut content = String::new();
+                    footnote(parser, &mut content, toc_builder, shorter, def.as_ref().to_owned(),
+                             id);
+                    let cur_len = parser.footnotes.len() + 1;
+                    parser.footnotes.push(format!("<li id=\"ref{}\">{}<a href=\"#supref{0}\" \
+                                                   rev=\"footnote\">↩</a></li>",
+                                                  cur_len, content));
```

Comment:
This is wrong. For example:

    foot[^note1], foot[^note2]
    
    [^note1]: text1

    [^note2]: text2

Will give the following event sequence:

    [
        Start(Paragraph),
        Text("foot"),
        FootnoteReference("note1"),
        Text(", foot"),
        FootnoteReference("note2"),
        End(Paragraph),
        Start(FootnoteDefinition("note1")),
        Start(Paragraph),
        Text("text1"),
        End(Paragraph),
        End(FootnoteDefinition("note1"))
        Start(FootnoteDefinition("note2")),
        Start(Paragraph),
        Text("text2"),
        End(Paragraph),
        End(FootnoteDefinition("note2"))
    ]

Correct me if I am wrong, but according to this code snippet, both footnote definitions will have the same id `cur_len`.